src/ui/reportView/callReport.ts

```diff
@@ -109,54 +146,64 @@ function createCommand(
 }
 
 /**
- * Run the visualize command to generate the report, parse the output to return the python server command and status
+ * Run the visualize command to generate the report, parse the output and return the result
  *
  * @param command - the cargo kani | kani command to run --visualize
- * @returns - A promise of the python command and the status code; Promise<visualizeOutput>
+ * @returns - the result of executing the visualize command and parsing the output
  */
 async function runVisualizeCommand(command: string, harnessName: string): Promise<visualizeOutput> {
 	try {
 		vscode.window.showInformationMessage(`Generating viewer report for ${harnessName}`);
 		const { stdout, stderr } = await execPromise(command);
-		const serveReportCommand: string = await parseReportOutput(stdout);
-		if (serveReportCommand === '') {
-			return { statusCode: 2, serverCommand: stderr };
+		const parseResult = await parseReportOutput(stdout);
+		if (parseResult === undefined) {
+			return { statusCode: 2, result: undefined };
 		}
 		console.error(`stderr: ${stderr}`);
 
-		return { statusCode: 0, serverCommand: serveReportCommand };
+		return { statusCode: 0, result: parseResult };
 	} catch (error) {
 		console.error(`exec error: ${error}`);
-		return { statusCode: 1, serverCommand: error as string };
+		return { statusCode: 1, result: undefined, error: error as string};
 	}
 }
 
 /**
- * Search for the python command contained in Kani's output and throw it onto the user's terminal
+ * Search for the path to the report printed in Kani's output, and detect if we are in a remote
+ * enviroment before returning the result.
  *
- * @param stdout - kani stdout output that contains the full python command to be run by the terminal
- * @returns - python command that looks like "python3 -m http.server --path path"
+ * @param stdout - Kani's standard output after running the visualize command
+ * @returns - undefined (error) or a result that indicates if the extension is executed on a local
+ *  or remote environment. The result includes a `path` (if local) or a `command` (if remote).
  */
-async function parseReportOutput(stdout: string): Promise<string> {
+async function parseReportOutput(stdout: string): Promise<visualizeResult | undefined> {
 	const kaniOutput: string = stdout;
 	const kaniOutputArray: string[] = kaniOutput.split('\n');
-	const searchString: string = 'python3';
+	const searchString: string = 'Report written to:';
 
 	for (const outputString of kaniOutputArray) {
-		if (outputString.includes(searchString)) {
-			const command: string = outputString.split(searchString)[1];
+		if (outputString.startsWith(searchString)) {
+			const reportPath: string = outputString.split(' ')[3]
```

Comment:
Paths can contain spaces, especially on windows. Since the beginning of the string is known constant, I suggest

```js
const reportPath: string = outputString.substring(searchString.length)
```

Reply:
Right, didn't think about that! Changed to use `substring` (one space added to the end of `searchString` so we don't need to trim `reportPath`).